Shared/Samples/Augment reality to collect data/AugmentRealityToCollectDataView.swift

```diff
@@ -0,0 +1,210 @@
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct AugmentRealityToCollectDataView: View {
+    /// A scene with an imagery basemap.
+    @State private var scene: ArcGIS.Scene = {
+        // Creates an elevation source from Terrain3D REST service.
+        let elevationServiceURL = URL(
+            string: "https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"
+        )!
+        let elevationSource = ArcGISTiledElevationSource(url: elevationServiceURL)
+        let surface = Surface()
+        surface.addElevationSource(elevationSource)
+        surface.backgroundGrid.isVisible = false
+        // Allow camera to go beneath the surface.
+        surface.navigationConstraint = .unconstrained
+        let scene = Scene(basemapStyle: .arcGISImagery)
+        scene.baseSurface = surface
+        scene.baseSurface.opacity = 0
+        return scene
+    }()
+    /// The AR tree survey service feature table.
+    private let featureTable = ServiceFeatureTable(
+        url: URL(string: "https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/AR_Tree_Survey/FeatureServer/0")!
+    )
+    /// The graphics overlay which shows marker symbols.
+    @State private var graphicsOverlay: GraphicsOverlay = {
+        let graphicsOverlay = GraphicsOverlay()
+        let tappedPointSymbol = SimpleMarkerSceneSymbol(
+            style: .diamond,
+            color: .orange,
+            height: 0.5,
+            width: 0.5,
+            depth: 0.5,
+            anchorPosition: .center
+        )
+        graphicsOverlay.renderer = SimpleRenderer(symbol: tappedPointSymbol)
+        graphicsOverlay.sceneProperties.surfacePlacement = .absolute
+        return graphicsOverlay
+    }()
+    /// The status text displayed to the user.
+    @State private var statusText = "Tap to create a feature"
+    /// The selected tree health for the new feature.
+    @State private var treeHealth: TreeHealth?
+    /// A Boolean value indicating whether a feature can be added .
+    @State private var canAddFeature = false
+    /// A Boolean value indicating whether the tree health action sheet is presented.
+    @State private var treeHealthSheetIsPresented = false
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    init() {
+        let featureLayer = FeatureLayer(featureTable: featureTable)
+        featureLayer.sceneProperties.surfacePlacement = .absolute
+        scene.addOperationalLayer(featureLayer)
```

Comment:
I missed this in my review of #353, but this should really be done in a model object so that it only happens once, not each time the parent's body is computed. Either that, or this should be done when `scene` is initialized and `featureTable` should be turned into a computed property.

Reply:
I added a model object.